pkg/telemetryservice/sql/telemetryservicesql.go

```diff
@@ -42,3 +50,32 @@ func BindSQLServiceHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Error to send to server", http.StatusBadRequest)
 	}
 }
+
+func injectSecret(setting *v1alpha1.SQLConnectionSetting) {
+	if setting == nil {
+		logger.Warn("empty telemetry service setting.")
+		return
+	}
+	if setting.Secret == nil {
+		logger.Warn("empty secret setting.")
+		return
+	}
+	secret, err := utils.GetSecret(*setting.Secret)
+	if err != nil {
+		logger.Errorf("unable to get secret for telemetry %v, error: %v", *setting.Secret, err)
+		return
+	}
+	var exist bool
+	*setting.Secret, exist = secret[PasswordSecretField]
+	if !exist {
```

Comment:
this code is repeating, can we make a method that retrieve the username/password from secret and then if success assign them to the MQTTSetting and SQLSetting?

Reply:
The [MQTT setting](https://github.com/FFFFFaraway/shifu/blob/9d478b7bd3f0146b9d9a53044797c179212a46b0/pkg/k8s/api/v1alpha1/edgedevice_types.go#L27) and [SQL setting](https://github.com/FFFFFaraway/shifu/blob/f2de96294c621c02db45afbbd2a6d3d35fa1c4cb/pkg/k8s/api/v1alpha1/telemetryservice_types.go#L95) structures are different. The SQL setting has a username field, but MQTT setting does not.